### PR TITLE
Tests for #119

### PR DESCRIPTION
**Pull request by Kranosos**
This branch focuses on adding Tests and Test-classes for all champion related classes and methods. 
Some minor fixes to the general structure of the project were also made.

Overall, I hope, that my team is contempt with the overall increase in code coverage!

### CODE REVIEW

**Comment by Ne0Strix** [comment]
Nice to see the long-awaited tests for the champions, and such odour-less code!

LGTM 😎

**Review by Ne0Strix — APPROVED** [review]

File: app/src/main/java/at/vunfer/openrealms/MainActivity.java [hunk 1]
```diff
@@ -256,11 +260,15 @@ public void run() {
...
+                                        defeatImage = findViewById(R.id.defeat_image);
                                         defeatImage.setVisibility(View.VISIBLE);
                                         defeatImage.getParent().bringChildToFront(defeatImage);
+                                        victoryImage = findViewById(R.id.victory_image);
+                                        victoryImage.setVisibility(View.INVISIBLE);
+                                        endTurnButton = findViewById(R.id.end_turn_button);
+                                        endTurnButton.setVisibility(View.INVISIBLE);
                                         Button endTurnButton = findViewById(R.id.end_turn_button);
-                                        endTurnButton.setVisibility(View.GONE);
+                                        endTurnButton.setVisibility(View.INVISIBLE);
```
Comment (Benjamin-Dippold) [inline app/src/main/java/at/vunfer/openrealms/MainActivity.java:271]: I think there was a merge conflict here, I doubt that we need to find and set the button to Invisible twice.

File: app/src/main/java/at/vunfer/openrealms/MainActivity.java [hunk 2]
```diff
@@ -273,14 +281,18 @@ public void run() {
...
-                                        endTurnButton.setVisibility(View.GONE);
+                                        endTurnButton = findViewById(R.id.end_turn_button);
+                                        endTurnButton.setVisibility(View.INVISIBLE);
                                         for (CardView c : cardViews) {
                                             c.setFaceDown();
                                         }
+                                        defeatImage = findViewById(R.id.defeat_image);
+                                        defeatImage.setVisibility(View.INVISIBLE);
+                                        endTurnButton = findViewById(R.id.end_turn_button);
+                                        endTurnButton.setVisibility(View.INVISIBLE);
```
Comment (Benjamin-Dippold) [inline app/src/main/java/at/vunfer/openrealms/MainActivity.java:295]: I think there was a merge conflict here, I doubt that we need to find and set the button to Invisible twice.

**Review by Benjamin-Dippold — CHANGES_REQUESTED** [review]

File: app/src/main/java/at/vunfer/openrealms/MainActivity.java (hunk 2 above)
Comment (Kranosos) [reply app/src/main/java/at/vunfer/openrealms/MainActivity.java:291]: Must have been it yea! Should be good now.

**Review by Benjamin-Dippold — APPROVED** [review]
LGTM

**Pull request merged**